### PR TITLE
Remove AsyncLazy<T>

### PR DESCRIPTION
**Pull request by martincostello**
* Remove `AsyncLazy<T>`.
  * Remove unused using statement.

### CODE REVIEW

File: JustSaying/AwsTools/AsyncLazy.cs [hunk 1]
```diff
@@ -1,10 +1,11 @@
 using System;
-using System.Runtime.CompilerServices;
 using System.Threading.Tasks;
 
 namespace JustSaying.AwsTools
 {
-    public class AsyncLazy<T> : Lazy<Task<T>>
+    // TODO Does this still meet async best-practice?
```
Comment (martincostello) [inline JustSaying/AwsTools/AsyncLazy.cs]: The usage of `Task.Factory.StartNew()` prompted this question.

Comment (slang25) [reply JustSaying/AwsTools/AsyncLazy.cs]: Good shout 👍 

This code originates from a time where `Task.Run` didn't exist ([see here](https://blogs.msdn.microsoft.com/pfxteam/2011/01/15/asynclazyt/)), it's correct in this implementation, but would be easier to reason with by using `Task.Run` here.

However as pointed out in the post where this code comes from, we could chose to not start a new task if we don't need the initial synchronous work in the delegate to be started on a different thread, and looking at the usage we don't need that, so I created a PR:
https://github.com/martincostello/JustSaying/pull/1

**Comment by AnthonySteele** [comment]
I think that this class basically does nothing now, it looks like an alias for `Lazy<Task<T>>`

**Comment by slang25** [comment]
Yeah, you could very well inline it, it's only used in one place

**Comment by slang25** [comment]
https://github.com/martincostello/JustSaying/pull/2

**Pull request merged**